main/src/ca/uwaterloo/flix/language/ast/shared/Mutability.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2025 Casper Dalgaard Nielsen
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.language.ast.shared
+
+sealed trait Mutability
+
+/**
+  * Represents whether a struct is mutable.
```

Comment:
Technically this comment should go on the trait (because thats the type one will  be hovering on)